### PR TITLE
Add last_updated tooltip to more info dialog

### PR DESCRIPTION
**Pull request by pszafer**
## Proposed change

Currently entities shows only `last_changed` of state.
If person has state `away` for a long time (during work or travel somewhere etc.) it's impossible to say from ui when location was last updated and if it is still "valid".
It might be useful for all entities.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Review by ludeeus — CHANGES_REQUESTED** [review]
We should not inject state properties in the state attributes.

**Comment by pszafer** [comment]
@ludeeus I changed it.
any thoughts how can I reuse css classes from `HaAttributes`?

**Comment by ludeeus** [comment]
Look at the styles used in that element, copy+paste -> adjust.

**Comment by pszafer** [comment]
I copied them. I thought that maybe is more efficient way.

File: src/dialogs/more-info/controls/more-info-person.ts [hunk 1]
```diff
@@ -28,12 +29,22 @@ class MoreInfoPerson extends LitElement {
     if (!this.hass || !this.stateObj) {
       return html``;
     }
-
     return html`
       <ha-attributes
         .stateObj=${this.stateObj}
         extraFilters="id,user_id,editable"
```
Comment (bramkragten) [inline src/dialogs/more-info/controls/more-info-person.ts:35]:
```suggestion
        extra-filters="id,user_id,editable"
```

**Comment by bramkragten** [comment]
I think we should do this differently, lets add it as a tooltip to the last changed time.

**Comment by pszafer** [comment]
by tooltip do you mean tooltip to hover over last_changed?

**Comment by bramkragten** [comment]
Yes

**Comment by pszafer** [comment]
but then it wouldn't be mobile accessible, would it?

**Comment by bramkragten** [comment]
It would show on tap on mobile

**Comment by zsarnett** [comment]
I don't think a tool tip is a good alternative here. It hides this information which I believe is important. We should just add it be a more user friend attribute. 

`Location Last Update: 45 min ago`

Under the state card content. 

This gives the user looking at this information a good representation of how accurate the zone may be.

**Comment by bramkragten** [comment]
This not just for person, adding it as a tooltip would make it available for every entity.

**Comment by pszafer** [comment]
ok, I will do it.

File: src/components/entity/state-info.js [hunk 2]
```diff
@@ -57,6 +57,27 @@ class StateInfo extends PolymerElement {
...
+          width: 200px;
+          transition: opacity 0.15s ease-in-out;
+          display: flex;
+          flex-direction: row;
+          justify-content: space-between;
+        }
+
+        .time-ago:hover + .last-updated {
+          opacity: 1;
+        }
```
Comment (bramkragten) [inline src/components/entity/state-info.js:80]:
```suggestion
```

File: src/components/entity/state-info.js [hunk 3]
```diff
@@ -79,6 +100,15 @@ class StateInfo extends PolymerElement {
...
           </div>
+          <div class="last-updated">
+            <div class="key">Last updated</div>
+            <div class="value">
+              <ha-relative-time
+                hass="[[hass]]"
+                datetime="[[stateObj.last_updated]]"
+              ></ha-relative-time>
+            </div>
+          </div>
```
Comment (bramkragten) [inline src/components/entity/state-info.js:111]:
```suggestion
            <paper-tooltip animation-delay="0">
              [[localize('ui.dialogs.more_info_control.last_updated')]]:
              <ha-relative-time
                hass="[[hass]]"
                datetime="[[stateObj.last_updated]]"
              ></ha-relative-time>
            </paper-tooltip>```

Comment (pszafer) [reply src/components/entity/state-info.js]: I put it in span, so it won't be centered horizontally over screen, but appear below `ha-relative-time`

File: src/components/entity/state-info.js [hunk 4]
```diff
@@ -74,10 +75,19 @@ class StateInfo extends PolymerElement {
 
         <template is="dom-if" if="[[inDialog]]">
           <div class="time-ago">
-            <ha-relative-time
-              hass="[[hass]]"
-              datetime="[[stateObj.last_changed]]"
-            ></ha-relative-time>
+            <span>
```
Comment (bramkragten) [inline src/components/entity/state-info.js:78]: Instead of the span, give it an id and use `for`

File: src/components/entity/state-info.js [hunk 5]
```diff
@@ -74,10 +75,19 @@ class StateInfo extends PolymerElement {
...
+              <ha-relative-time
+                hass="[[hass]]"
+                datetime="[[stateObj.last_changed]]"
+              ></ha-relative-time>
+              <paper-tooltip animation-delay="0">
+                [[localize('ui.dialogs.more_info_control.last_updated')]]:
```
Comment (bramkragten) [inline src/components/entity/state-info.js:84]: This string doesn't exist yet? You have to add it to `src/translations/en.json`

Comment (pszafer) [reply src/components/entity/state-info.js]: added. one notice, even if string is added it doesn't show in my browser. Actually it doesn't show even if I use some other string like `ui.dialogs.more_info_control.dismiss`.

File: src/components/entity/state-info.js [hunk 6]
```diff
@@ -71,13 +72,23 @@ class StateInfo extends PolymerElement {
         <div class="name" in-dialog$="[[inDialog]]">
           [[computeStateName(stateObj)]]
         </div>
-
         <template is="dom-if" if="[[inDialog]]">
           <div class="time-ago">
             <ha-relative-time
+              id="ha_relative_time_[[stateObj.context.id]]"
```
Comment (bramkragten) [inline src/components/entity/state-info.js:78]: The id can just be `last_changed`

**Review by bramkragten — APPROVED** [review]

**Comment by bramkragten** [comment]
Oh, this is missing the LocalizeMixin...

**Comment by bramkragten** [comment]
Fixed in dev.

**Pull request merged**